src/synapse/dataProvider.js

```diff
@@ -57,6 +57,24 @@ const resourceMap = {
     }),
     data: "connections",
   },
+  servernotices: {
```

Comment:
Since this is not a typical endpoint (with CRUD), we could as well export a separate function to handle just this one.
E.g.
`export const serverNotice = (user_ids, message) => { ... }`
and in ServerNotices.js you can
`import { serverNotice } from "../synapse/dataProvider.js";`

Reply:
Do you think a seperate function is the right solution?

> export const serverNotice = (user_ids, message) => { ... }

My suggestion would be an additional method for the dataProvider, e.g "sendMessage"

> Tip: If your API supports more request types, you can add more methods to the Data Provider (for instance to support upserts, aggregations, or Remote Procedure Call). React-admin won’t call these methods directly, but you can call them in your own component thanks to the useDataProvider hook described in the Querying the API documentation.

https://marmelab.com/react-admin/DataProviders.html#writing-your-own-data-provider

Similar is necessary for: https://github.com/matrix-org/synapse/blob/master/docs/admin_api/shutdown_room.md

Reply:
Agreed, we can do that